filemanager.lua

```diff
@@ -1095,9 +1113,13 @@ function preMousePress(view, event)
 		local x, y = event:Position()
 		-- Fixes the y because softwrap messes with it
 		local new_x, new_y = tree_view:GetMouseClickLocation(x, y)
-		-- Try to open whatever is at the click's y index
-		-- Will go into/back dirs based on what's clicked, nothing gets expanded
-		try_open_at_y(new_y)
+
+		--open item if it is selected, otherwise select it.
+		if tree_view.Buf.Cursor.Loc.Y == new_y then
```

Comment:
I personally prefer double click but it also solves the problem that if you use a mouse click to switch to the tree pane it will enter whatever directory the cursor is on

Reply:
That problem could be solved without requiring double-click, I think?

Possibly by adding in a section to the `pre` callbacks to check for `if view ~= tree_view`, getting cursor xy position, then checking if it's anywhere inside the dimensions of the tree. If it is, you could set the current view to the `tree_view`, then perform the scroll action.

But I'm not exactly sure how you'd get the xy of the tree. Probably possible..